src/components/NbtscanTool/NbtscanTool.tsx

```diff
@@ -32,17 +20,39 @@ interface FormValuesType {
     retransmits: number;
 }
 
-//sets the state of the tool; loading or not, what the output is
+/**
+ * The NbtscanTool component.
+ * @returns The NbtscanTool component.
+ */
 const NbtscanTool = () => {
-    const [loading, setLoading] = useState(false);
-    const [output, setOutput] = useState("");
-    const [checkedAdvanced, setCheckedAdvanced] = useState(false);
-    const [pid, setPid] = useState("");
-    const [allowSave, setAllowSave] = useState(false);
-    const [hasSaved, setHasSaved] = useState(false);
-    //const [subnet, setsubnet] = useState("");
-    const [checkedPacketDump, setcheckedPacketDump] = useState(false);
-
+    // Component State Variables.
+    const [loading, setLoading] = useState(false); // State variable to control the loading overlay
+    const [output, setOutput] = useState(""); // State variable to store the output from the nbtscan command
+    const [checkedAdvanced, setCheckedAdvanced] = useState(false); // State variable to track if the advanced mode is enabled
+    const [pid, setPid] = useState(""); // State variable to store the process ID (PID) of the running command
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [allowSave, setAllowSave] = useState(false); // State variable to control whether the output can be saved
+    const [hasSaved, setHasSaved] = useState(false); // State variable to track if the output has already been saved
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the modal is opened.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state of the modal.
+    const [subnet, setSubnet] = useState(""); // State variable to store the entered subnet value
```

Comment:
Sorry mate. Forgot to edit it out. It was supposed to be removed as I think the code uses the subnet value from form instead of the state variable. It was commented in the code before.

Reply:
No worries. I think it looks good.